Update highlights

0xAdk opened the pull request:
* Add `@type.builtin` highlight to scalar types

* Fix `@variable` highlight having a higher priority than `@type` for `identifier`s in `type_declaration`

This is dependent on #5 otherwise tree-sitter will error `Query error at 6:41. Invalid node type f16`

0xAdk committed:
Update highlights

* Add `@type.builtin` highlight to builtin scalar types

* Fix `@variable` highlight priority

* Add `@punctuation.delimiter` highlight to `->`

szebniok commented:
Thanks, I can confirm that the `@type.builtin` highlights work properly.
Are you sure that the `@variable` highlighting issue was actually fixed by this change? I could only get it to work if I moved the first line (`(identifier) @variable`) below the `(type_declaration (identifier) @type)` line. I was testing it using the `tree-sitter highlight` command. I couldn't find any bit of information that would say something about the order of the queries.

0xAdk commented:
I was basing it off how it looks in neovim. After a bit of testing with tree-sitter highlight It looks like they use two different systems to specify priority. I also found this tree-sitter issue tree-sitter/tree-sitter/issues/1374 which seems to be the same problem.

what I understand from testing neovim, it prioritizes smaller highlights, so in the case of 
```query
(identifier) @variable
(type_declaration) @type
```
`@variable` wins. But if two queries have the same size it uses the one that came last, so in the case of
```
(identifier) @variable
(type_declaration (identifier) @type)
```
`@type` wins.

from what I understand from tree-sitter highlight it looks like it does the exact opposite. I'm not sure.
In a test, the larger highlight always won regardless of if it came before or after another query
```lisp
(variable_identifier_declaration) @comment
(identifier) @variable
; and
(identifier) @variable
(variable_identifier_declaration) @comment
```
`@comment`  wins in both cases.
When testing for which highlight has priority when they have the same size
```
; @variable wins
(identifier) @variable
(type_declaration (identifier) @type)
; @type wins
(type_declaration (identifier) @type)
(identifier) @variable
```
the query that came first got the priority.

also, through this I found out the current `(attribute (identifier) @attribute)` highlight at the bottom of the file only works for neovim, not tree-sitter highlight.

---
I'll rework the pr to be tree-sitter highlight specific, and move the `@type` priority thing to a nvim-treesitter pr

0xAdk commented:
changed the commit to

* Add `@type.builtin` highlight to builtin scalar types

* Fix `@variable` highlight priority

* Add `@punctuation.delimiter` highlight to `->`

added the `->` highlight part to keep it in sync with nvim-treesitter

0xAdk commented:
Previously a whole bunch of highlights were getting covered up due to `@variable` being defined before them.

---
top is this commit, bottom is master
![highlight difference](https://user-images.githubusercontent.com/29005635/210501609-639dfa54-85ef-4af1-989d-505564c96409.png)

szebniok commented:
Thank you for your extensive investigation!
